common/lib/xmodule/xmodule/js/src/video/09_events_plugin.js

```diff
@@ -0,0 +1,141 @@
+     * @param {Object} options
+     * @return {jquery Promise}
+     */
+    var EventsPlugin = function(state, i18n, options) {
+        if (!(this instanceof EventsPlugin)) {
+            return new EventsPlugin(state, i18n, options);
+        }
+
+        _.bindAll(this, 'onReady', 'onPlay', 'onPause', 'onEnded', 'onSeek',
+            'onSpeedChange', 'onShowLanguageMenu', 'onHideLanguageMenu', 'onSkip',
+            'onShowCaptions', 'onHideCaptions', 'destroy');
+        this.state = state;
+        this.options = _.extend({}, options);
+        this.state.videoEventsPlugin = this;
+        this.i18n = i18n;
+        this.initialize();
+
+        return $.Deferred().resolve().promise();
+    };
+
+    EventsPlugin.moduleName = 'EventsPlugin';
+    EventsPlugin.prototype = {
+        destroy: function () {
+            this.state.el.off({
+                'ready': this.onReady,
+                'play': this.onPlay,
+                'pause': this.onPause,
+                'ended stop': this.onEnded,
+                'seek': this.onSeek,
+                'skip': this.onSkip,
+                'speedchange': this.onSpeedChange,
+                'language_menu:show': this.onShowLanguageMenu,
+                'language_menu:hide': this.onHideLanguageMenu,
+                'captions:show': this.onShowCaptions,
+                'captions:hide': this.onHideCaptions,
+                'destroy': this.destroy
+            });
+            delete this.state.videoEventsPlugin;
+        },
+
+        initialize: function() {
+            this.bindHandlers();
+        },
+
+        bindHandlers: function() {
+            this.state.el.on({
+                'ready': this.onReady,
+                'play': this.onPlay,
+                'pause': this.onPause,
+                'ended stop': this.onEnded,
+                'seek': this.onSeek,
+                'skip': this.onSkip,
+                'speedchange': this.onSpeedChange,
+                'language_menu:show': this.onShowLanguageMenu,
+                'language_menu:hide': this.onHideLanguageMenu,
+                'captions:show': this.onShowCaptions,
+                'captions:hide': this.onHideCaptions,
+                'destroy': this.destroy
+            });
+        },
+
+        onReady: function () {
+            this.log('load_video');
+        },
+
+        onPlay: function () {
+            this.log('play_video', {currentTime: this.getCurrentTime()});
+        },
+
+        onPause: function () {
+            this.log('pause_video', {currentTime: this.getCurrentTime()});
+        },
+
+        onEnded: function () {
+            this.log('stop_video', {currentTime: this.getCurrentTime()});
+        },
+
+        onSkip: function (event, doNotShowAgain) {
+            var info = {currentTime: this.getCurrentTime()},
+                eventName = doNotShowAgain ? 'do_not_show_again_video': 'skip_video';
+            this.log(eventName, info);
+        },
```

Comment:
it emits `do_not_show_again_video ` event when the user clicks on `Do not show again` button.
It emits `skip_video ` event when the user clicks on `skip` button.